gocd/bci_repo_publish.py

```diff
@@ -131,13 +134,14 @@ def run(self, version, token=None):
             return
 
         # Check openQA results
-        openqa_passed = True
         for pkg in packages:
             passed = 0
             pending = 0
             failed = 0
+            openqa_tested = False
```

Comment:
Variable is unnecessary, there is already passed/pending/failed.

What about this:

```python3
         # Check openQA results
        for pkg in packages:
            passed = 0
            pending = 0
            failed = 0
            for job in self.openqa_jobs_for_product(arch=pkg['arch'], version=version,
                                                    build=pkg['built_version']):
                if job['result'] in ('passed', 'softfailed'):
                    passed += 1
                elif job['result'] == 'none':
                    self.logger.info(f'https://openqa.suse.de/tests/{job["id"]} pending')
                    pending += 1
                else:
                    self.logger.warning(f'https://openqa.suse.de/tests/{job["id"]} failed')
                    failed += 1

            if pkg['arch'] in ('ppc64le', 's390x') and failed == 0:
                openqa_passed_packages.append(pkg)
            elif passed > 0 and pending == 0 and failed == 0:
                openqa_passed_packages.append(pkg)
            else:
                self.logger.info(f'openQA did not (yet) pass for {pkg["name"]}: {passed}/{pending}/{failed}')
```

Reply:
only testing for "failed == 0" means that you can't distinguish between pending != 0 and pending == 0 (which means not yet scheduled). the boolean finds that there was at least one job. 

you can emulate the boolean by testing for `passed + failed + pending` if you think we need to save on variables.

Reply:
`if pkg['arch'] in ('ppc64le', 's390x') and failed == 0 and pending == 0:`?

Reply:
that detects "no job for ppc64 or s390x" but you don't detect whether its because openqa hasn't scheduled it yet or because openqa will never schedule it. 

thats why I inverted the logic to check for mandatory arches (defined as aarch64 and x86_64) and only move forward with releasing s390x and ppc64 if the mandatory arches have passed *and* there are no fails and no pending for ppc64 and s390x.